File: 2529-range-product-queries-of-powers/2529-range-product-queries-of-powers.py
class Solution:
    def productQueries(self, num: int, queries: List[List[int]]) -> List[int]:
        powers = []
        while num:
            lowBit = num & -num
            powers.append(lowBit)
            num ^= lowBit

        res = []
        for x, y in queries:
            val = 1
            for i in range(x, y + 1):
                val *= powers[i]
            res.append(val % (10 ** 9 + 7))
        return res
        

File: 2529-range-product-queries-of-powers/2529-range-product-queries-of-powers.js
/**
 * @param {number} n
 * @param {number[][]} queries
 * @return {number[]}
 */
var productQueries = function(n, queries) {
    const powers = [];
    while (n) {
        const b = n & -n;
        powers.push(b);
        n ^= b;
    }

    const res = [];
    for (const [x, y] of queries) {
        let val = 1;
        for (let i = x; i < y + 1; i++) {
            val *= powers[i];
        }
        res.push(val % (10**9 + 7));
    }
    return res;
};